lib/write-sources.js

```diff
@@ -22,6 +23,12 @@ const writeSources = (filename, content, outdir, options) => {
 
   const outputFilepath = path.join(outdir, filename);
 
+  if (options.preserve) {
+    mkdirp.sync(
```

Comment:
Use https://github.com/jprichardson/node-fs-extra/blob/HEAD/docs/ensureDir.md, so that you don't need to add new dependency